pkg/doc/verifiable/credential_jwt.go

```diff
@@ -52,107 +53,78 @@ func newJWTCredClaims(vc *Credential, minimizeVc bool) (*JWTCredClaims, error) {
+		return nil, err
 	}
 
-	// make the merge
-	for k, v := range rawMap {
-		jsonCred[k] = v
+	credClaims := &JWTCredClaims{
+		Claims: jwtClaims,
+		VC:     vcMap,
 	}
-
-	return nil
-}
-
-// jwtCredClaimsDecoder parses JWT claims from serialized token into JWTCredClaims struct and JSON object.
-// The implementation depends on the type of serialization, e.g. signed (JWT), unsecured (plain JWT).
-type jwtCredClaimsDecoder interface {
-	UnmarshalClaims(rawJwt []byte) (*JWTCredClaims, error)
-	UnmarshalVCClaim(rawJwt []byte) (map[string]interface{}, error)
+	return credClaims, nil
 }
 
-// decodeCredJWT parses JWT from the specified bytes array in compact format using jwtDecoder.
-// It returns decoded Verifiable Credential refined by JWT Claims in raw byte array and rawCredential form.
-func decodeCredJWT(rawJWT []byte, credClaimsDecoder jwtCredClaimsDecoder) ([]byte, *rawCredential, error) {
-	credClaims, err := credClaimsDecoder.UnmarshalClaims(rawJWT)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to decode Verifiable Credential JWT claims: %w", err)
-	}
+// JWTCredClaimsUnmarshaller unmarshals verifiable credential bytes into JWT claims with extra "vc" claim.
+type JWTCredClaimsUnmarshaller func(vcJWTBytes []byte) (*JWTCredClaims, error)
 
-	credJSON, err := credClaimsDecoder.UnmarshalVCClaim(rawJWT)
+// decodeCredJWT parses JWT from the specified bytes array in compact format using unmarshaller.
+// It returns decoded Verifiable Credential refined by JWT Claims in raw byte array form.
+func decodeCredJWT(rawJWT []byte, unmarshaller JWTCredClaimsUnmarshaller) ([]byte, error) {
+	credClaims, err := unmarshaller(rawJWT)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to decode raw Verifiable Credential JWT claims: %w", err)
+		return nil, fmt.Errorf("failed to decode Verifiable Credential JWT claims: %w", err)
 	}
 
 	// Apply VC-related claims from JWT.
 	credClaims.refineFromJWTClaims()
-	// Complement original "vc" JSON claim with data refined from JWT claims.
-	if err = mergeRefinedVC(credJSON, credClaims.Credential); err != nil {
-		return nil, nil, fmt.Errorf("failed to merge refined VC: %w", err)
-	}
 
 	var vcData []byte
-	if vcData, err = json.Marshal(credJSON); err != nil {
-		return nil, nil, errors.New("failed to marshal 'vc' claim of JWT")
+	if vcData, err = json.Marshal(credClaims.VC); err != nil {
+		return nil, errors.New("failed to marshal 'vc' claim of JWT")
 	}
 
-	return vcData, credClaims.Credential, nil
+	return vcData, nil
 }
 
 func (jcc *JWTCredClaims) refineFromJWTClaims() {
-	raw := jcc.Credential
+	vcMap := jcc.VC
+	claims := jcc.Claims
 
-	if iss := jcc.Issuer; iss != "" {
-		refineVCIssuerFromJWTClaims(raw, iss)
+	if iss := claims.Issuer; iss != "" {
+		refineVCIssuerFromJWTClaims(vcMap, iss)
 	}
 
-	if nbf := jcc.NotBefore; nbf != nil {
+	if nbf := claims.NotBefore; nbf != nil {
 		nbfTime := nbf.Time().UTC()
-		raw.Issued = &nbfTime
+		vcMap["issuanceDate"] = nbfTime.Format(time.RFC3339)
```

Comment:
some of these strings could make sense as named constants.

Reply:
Agree! Done.